packages/client/src/client4.ts

```diff
@@ -1928,11 +1928,30 @@ export default class Client4 {
         );
     };
 
-    getPostThread = (postId: string, fetchThreads = true, collapsedThreads = false, collapsedThreadsExtended = false) => {
-        return this.doFetch<PostList>(
-            `${this.getPostRoute(postId)}/thread${buildQueryString({skipFetchThreads: !fetchThreads, collapsedThreads, collapsedThreadsExtended})}`,
+    getPostThread = async (postId: string, fetchThreads = true, collapsedThreads = false, collapsedThreadsExtended = false, direction: 'up'|'down' = 'down', perPage: number = PER_PAGE_DEFAULT, fromCreateAt?: number, fromPost?: string, prevList?: PostList): Promise<PostList> => {
```

Comment:
I'm not looking forward to adding another 4 arguments to this since that's getting hard to read. I tried using an argument object to make that easier to use, but since we'd want to maintain backwards compatibility with the old arguments because this is a public API. Unfortunately, it ends up being more than a little ugly with how I was thinking of doing it.

```typescript
    getPostThread(postId: string): Promise<PaginatedPostList>;
    getPostThread(postId: string, fetchThreads?: boolean, collapsedThreads?: boolean, collapsedThreadsExtended?: boolean): Promise<PaginatedPostList>;
    getPostThread(postId: string, options?: {fetchThreads?: boolean; collapsedThreads?: boolean; collapsedThreadsExtended?: boolean; direction?: 'up' | 'down'; perPage?: number; fromCreateAt?: number; fromPost?: string}): Promise<PaginatedPostList>;
    getPostThread(postId: string, ...otherArgs: any[]) {
        let fetchThreads = true;
        let collapsedThreads = false;
        let collapsedThreadsExtended = false;
        let direction = 'down';
        let perPage = PER_PAGE_DEFAULT;
        let fromCreateAt;
        let fromPost;
        if (otherArgs) {
            if (otherArgs.length > 1) {
                fetchThreads = otherArgs[0] ?? fetchThreads;
                collapsedThreads = otherArgs[1] ?? collapsedThreads;
                collapsedThreadsExtended = otherArgs[2] ?? collapsedThreadsExtended;
            } else if (otherArgs.length === 1) {
                fetchThreads = otherArgs[0].fetchThreads ?? fetchThreads;
                collapsedThreads = otherArgs[0].collapsedThreads ?? collapsedThreads;
                collapsedThreadsExtended = otherArgs[0].collapsedThreadsExtended ?? collapsedThreadsExtended;
                direction = otherArgs[0].direction ?? direction;
                perPage = otherArgs[0].perPage ?? perPage;
                fromCreateAt = otherArgs[0].fromCreateAt ?? fromCreateAt;
                fromPost = otherArgs[0].fromPost ?? fromPost;
            }
        }

        return this.doFetch<PaginatedPostList>(
            `${this.getPostRoute(postId)}/thread${buildQueryString({skipFetchThreads: !fetchThreads, collapsedThreads, collapsedThreadsExtended, direction, perPage, fromPost, fromCreateAt})}`,
            {method: 'get'},
        );
    }
```

Nothing to change here unless you have some better ideas for doing this, but it would be nice to figure out a way for us to solve this sort of thing going forward as some of these APIs get more complicated.

Reply:
well, there is only one option: Refactoring!
I would write a new function, maybe `getPaginatedPostThread` use 2 params, `rootId` and `options`. The good thing about that is, that you can always extend the `options` object with new properties without obfuscating the param list.

And to support the "old" function just wrap the new one into the old one! ;D
That way we can eventually deprecate it (create an alias with `getPostThread_DEPRECATED`) and remove it in a future major release.

WDYT?

Reply:
I do like the idea of adding a new method instead of trying to force backwards compatibility onto the existing one. I don't know about adding a deprecated alias since that seems a bit heavy handed for a module the size of Client4, but we can figure that out when we get around to doing that deprecation